x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CheckShrinkReadyStep.java

```diff
@@ -64,6 +73,18 @@ public Result isConditionMet(Index index, ClusterState clusterState) {
             throw new IllegalStateException("Cannot check shrink allocation as there are no allocation rules by _id");
         }
 
+        boolean nodeBeingRemoved = NodesShutdownMetadata.getShutdowns(clusterState)
+            .map(NodesShutdownMetadata::getAllNodeMetadataMap)
+            .map(nmm -> nmm.get(idShardsShouldBeOn))
+            .map(snsm -> snsm.getType() == SingleNodeShutdownMetadata.Type.REMOVE)
+            .orElse(false);
+
+        if (nodeBeingRemoved) {
+            completable = false;
+            return new Result(false, new SingleMessageFieldInfo("node with id [" + idShardsShouldBeOn +
+                "] is currently marked as shutting down for removal"));
+        }
```

Comment:
Hmm.. I would say they are both around the same, so maybe it's better to do the check afterwards. This would mean that we'd prevent shutdown for slightly longer, but avoid extra relocation if the allocation were already complete

Reply:
I moved this check down below and added a test for the differing behavior